docker/base-deps/Dockerfile

```diff
@@ -27,53 +27,64 @@ RUN apt-get update -y \
 USER $RAY_UID
 ENV HOME=/home/ray
 
-RUN sudo apt-get update -y && sudo apt-get upgrade -y \
+SHELL ["/bin/bash", "-c"]
+
+
+RUN  export arch=$(uname -m); \
+     if [ "$arch" == "aarm64" ]; then \
+	arch="arm64"; \
+     fi; \
+    sudo apt-get update -y && sudo apt-get upgrade -y \
     && sudo apt-get install -y \
-        git \
-        wget \
-        cmake \
-        g++ \ 
-        zlib1g-dev \
-        $(if [ "$AUTOSCALER" = "autoscaler" ]; then echo \
-        tmux \
-        screen \
-        rsync \
-        openssh-client \
-        gnupg; fi) \
-    && wget \
-        --quiet "https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh" \
-        -O /tmp/miniconda.sh \
+	git \
+	wget \
+	cmake \
+	gcc \
+	g++ \
+	make \
+	zlib1g-dev \
+	zlib1g \
+	$(if [ "$AUTOSCALER" == "autoscaler" ]; then echo \
+	tmux \
+	screen \
+	rsync \
+	openssh-client \
+	gnupg; fi) \
+     && wget \
+	--quiet "https://github.com/conda-forge/miniforge/releases/download/4.8.5-1/Miniforge3-4.8.5-1-Linux-${arch}.sh" \
+	-O /tmp/miniconda.sh \
     && /bin/bash /tmp/miniconda.sh -b -u -p $HOME/anaconda3 \
-    && $HOME/anaconda3/bin/conda init \ 
+    && $HOME/anaconda3/bin/conda init \
     && echo 'export PATH=$HOME/anaconda3/bin:$PATH' >> /home/ray/.bashrc \
     && rm /tmp/miniconda.sh \
-    && $HOME/anaconda3/bin/conda install -y \
-        libgcc python=$PYTHON_VERSION \
+    && $HOME/anaconda3/bin/conda install -y mamba \
```

Comment:
Can we stick to `pip` and `apt-get` for dependencies? They're admittedly old and boring, but reliable and well understood.

Reply:
Mamba is pretty common these days and it gives much better error messages when it can't resolve the packages (more likely to happen with ARM because less conda packages published).

That being said it is optional, eg the PR would work without it just more painful to debug long term.

Reply:
Oh also, just to be clear the existing Dockerfiles don't just use apt & pip, they already use conda, mamba pulls from conda-forge so it's not really a new source.